src/genesis.rs

```diff
@@ -0,0 +1,94 @@
+use {
+    crate::ValidatorType,
+    log::*,
+    rand::Rng,
+    solana_core::gen_keys::GenKeys,
+    solana_sdk::signature::{write_keypair_file, Keypair},
+    std::{error::Error, path::PathBuf, result::Result},
+};
+
+pub struct Genesis {
+    config_dir: PathBuf,
+    key_generator: GenKeys,
+}
+
+impl Genesis {
+    pub fn new(solana_root: &PathBuf) -> Self {
+        let config_dir = solana_root.join("config-k8s");
+        if config_dir.exists() {
+            std::fs::remove_dir_all(&config_dir).unwrap();
+        }
+        std::fs::create_dir_all(&config_dir).unwrap();
+
+        let seed: [u8; 32] = rand::thread_rng().gen();
+
+        Self {
+            config_dir,
+            key_generator: GenKeys::new(seed),
+        }
+    }
+
+    pub fn generate_faucet(&mut self) -> Result<(), Box<dyn Error>> {
+        info!("generating faucet keypair");
+        let outfile = self.config_dir.join("faucet.json");
+        let keypair = self.key_generator.gen_keypair();
+
+        write_keypair_file(&keypair, outfile)?;
+        Ok(())
+    }
+
+    pub fn generate_accounts(
+        &mut self,
+        validator_type: ValidatorType,
+        number_of_accounts: usize,
+    ) -> Result<(), Box<dyn Error>> {
+        if validator_type == ValidatorType::Client {
+            return Err("Client valdiator_type in generate_accounts not allowed".into());
+        }
+
+        info!("generating {number_of_accounts} {validator_type} accounts...");
+
+        let mut account_types = vec!["identity", "stake-account", "vote-account"];
+        match validator_type {
+            ValidatorType::Bootstrap | ValidatorType::Standard => (),
+            ValidatorType::RPC => {
+                account_types.pop(); // no vote-account for RPC
```

Comment:
nit: I guess I missed something here. why do we need a stake-account for an RPC node?

Reply:
you are correct. my mind here got mixed up between client and Rpc node. will fix